BeeTracker/app/src/main/java/com/blueprint/foe/beetracker/CaptureFragment.java

```diff
@@ -1,51 +1,183 @@
+import com.otaliastudios.cameraview.CameraUtils;
+import com.otaliastudios.cameraview.CameraView;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
 
 /**
  * This fragment will allow the user to take a picture in-app.
  */
-
 public class CaptureFragment extends Fragment {
     private static final String TAG = CaptureFragment.class.toString();
+    private CameraView mCameraView;
 
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         View view = inflater.inflate(R.layout.capture_fragment, container, false);
 
-        Button nextButton = (Button) view.findViewById(R.id.nextButton);
-        nextButton.setOnClickListener(new View.OnClickListener() {
+        TextView cancelButton = (TextView) view.findViewById(R.id.cancelButton);
+        cancelButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                Fragment newFragment = new IdentifyFragment();
-                FragmentTransaction transaction = getFragmentManager().beginTransaction();
-
-                // Replace whatever is in the fragment_container view with this fragment,
-                // and add the transaction to the back stack
-                transaction.replace(R.id.fragment_container, newFragment);
-                transaction.addToBackStack(null);
-
-                transaction.commit();
+                // Back to Home page activity
+                getActivity().finish();
             }
         });
 
-        Button cancelButton = (Button) view.findViewById(R.id.cancelButton);
-        cancelButton.setOnClickListener(new View.OnClickListener() {
+        mCameraView = (CameraView) view.findViewById(R.id.camera);
+
+        mCameraView.addCameraListener(new CameraListener() {
             @Override
-            public void onClick(View view) {
-                // Back to Home page activity
-                getActivity().finish();
+            public void onPictureTaken(byte[] picture) {
+                // CameraUtils will generate image, with correct EXIF orientation, in a worker thread.
+                CameraUtils.decodeBitmap(picture, new CameraUtils.BitmapCallback() {
+                    @Override
+                    public void onBitmapReady(Bitmap bitmap) {
+                        if (checkForPermissions(getActivity())) {
+                            String filename = null;
+                            try {
+                                filename = StorageAccessor.saveBitmapExternally(bitmap);
+                            } catch (FileNotFoundException e) {
+                                e.printStackTrace();
+                                errorAndExit("There was a problem saving your image externally.");
+                            }
+                            saveAndLaunchNextFragment(filename);
+                        } else {
```

Comment:
Comment would be nice, saying once permission is granted you'll read from internal memory and save externally